ml-agents-envs/mlagents_envs/environment.py

```diff
@@ -341,8 +341,13 @@ def set_actions(self, behavior_name: BehaviorName, action: np.ndarray) -> None:
         if behavior_name not in self._env_state:
             return
         spec = self._env_specs[behavior_name]
-        expected_type = np.float32 if spec.is_action_continuous() else np.int32
-        expected_shape = (len(self._env_state[behavior_name][0]), spec.action_size)
+        expected_type = (
```

Comment:
Could we push this logic into an ActionSpec method instead? That way we can reduce the amount of calls that we're saying are deprecated/legacy/backcompat (like is_action_continuous)